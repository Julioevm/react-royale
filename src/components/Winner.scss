@use "variables" as v;

.modal {
	position: fixed;
	left: 0;
	top: 0;
	right: 0;
	bottom: 0;
	background-color: rgba(0, 0, 0, 0.5);
	display: flex;
	align-items: center;
	justify-content: center;
}

.modal-card-container {
	perspective: 40em;
}

.modal-card {
	--bi: radial-gradient(ellipse at center, #446bb3 0%, #000000 90%);
	position: relative;
	width: 320px;
	height: 380px;
	padding: 3em;
	color: v.$text-color;
	transform: rotateY(30deg) rotateX(15deg);
	transform-style: preserve-3d;
	transition: transform 1s;
	text-align: center;

	&:hover {
		transform: rotateY(-15deg) rotateX(-15deg);
	}

	.winnerName {
		font-size: 1.5em;
	}

	.image {
		display: block;
		margin-top: 10px;
		margin-bottom: 10px;
		margin-left: auto;
		margin-right: auto;
		width: 50%;
		height: 220px;
		width: 250px;
		overflow: hidden;
		object-fit: cover;
	}

	p {
		margin: 5px;
	}
}

.layers {
	position: absolute;
	left: 0;
	top: 0;
	width: 100%;
	height: 100%;
	transform-style: preserve-3d;
	z-index: -1;
}

.layer {
	position: absolute;
	left: 0;
	top: 0;
	width: 100%;
	height: 100%;
	border-radius: 1em;
	background-image: var(--bi);
	transform: translateZ(var(--tz));
	box-shadow: 0 0 0.5em #000d inset;

	@for $i from 0 to 10 {
		&:nth-child(#{$i + 1}) {
			--tz: #{$i * -4}px;
		}
	}

	&:last-child {
		box-shadow: 0 0 0.5em #000d inset, 0 0 5px #000;
	}
}
