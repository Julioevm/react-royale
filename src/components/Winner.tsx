import React from "react";
import { Player } from "../DAL/Player";
import "./Winner.scss";

export default function Winner(props: {
	winner: Player | undefined;
	onClose: () => void;
	show: boolean;
}) {
	if (!props.show) {
		return null;
	}

	return (
		<div className="modal" onClick={props.onClose}>
			<div
				className="modal-card-container"
				onClick={(e) => e.stopPropagation()}
			>
				<div className="modal-card" data-testid={"modal-card"}>
					<h1>The winner is...</h1>
					<img className="image" src={props.winner?.image} alt="Winner portrait" />
					<p className="winnerName">{props.winner?.name}</p>
					<p>{props.winner?.state.name}</p>
					<p>{props.winner?.weapon.name}</p>
					<div className="layers">
						<div className="layer"></div>
						<div className="layer"></div>
						<div className="layer"></div>
						<div className="layer"></div>
						<div className="layer"></div>
					</div>
				</div>
			</div>
		</div>
	);
}
