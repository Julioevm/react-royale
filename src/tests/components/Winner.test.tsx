import React from "react";
import renderer from "react-test-renderer";
import Winner from "../../components/Winner";
import { player1 } from "../__fixtures__/players";

describe("<Winner/>", () => {
	it("renders", () => {
		const wrapper = renderer
			.create(<Winner winner={player1} show={true} onClose={() => {}} />)
			.toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
