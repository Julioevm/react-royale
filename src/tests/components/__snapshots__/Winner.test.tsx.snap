// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Winner/> renders 1`] = `
<div
  className="modal"
  onClick={[Function]}
>
  <div
    className="modal-card-container"
    onClick={[Function]}
  >
    <div
      className="modal-card"
      data-testid="modal-card"
    >
      <h1>
        The winner is...
      </h1>
      <img
        alt="Winner portrait"
        className="image"
      />
      <p
        className="winnerName"
      >
        Player1
      </p>
      <p>
        Healthy
      </p>
      <p>
        Fists
      </p>
      <div
        className="layers"
      >
        <div
          className="layer"
        />
        <div
          className="layer"
        />
        <div
          className="layer"
        />
        <div
          className="layer"
        />
        <div
          className="layer"
        />
      </div>
    </div>
  </div>
</div>
`;
